source/views/tasks_view.py

```diff
@@ -38,15 +39,18 @@ def post(self):
         data = request.json
         task_name = data.get('task_name', None)
         if not task_name:
-            return 400
+            raise NotFound('Task name is None')
```

Comment:
NotFound is 404 and should happens when the query to the database returns an empty set, when the request data is incomplete we use `400 BadRequest`